source/plugin/Assets/GoogleMobileAds/Platforms/Unity/RewardedAdClient.cs

```diff
@@ -0,0 +1,205 @@
+                if (OnAdClosed != null)
+                {
+                    OnAdClosed.Invoke(this, EventArgs.Empty);
+                }
+                AdBehaviour.ResumeGame();
+                if (OnUserEarnedReward != null)
+                {
+                    OnUserEarnedReward.Invoke(this, GetRewardItem());
+                }
+            });
+        }
+
+        private void CreateButtonBehavior()
+        {
+            buttonBehaviour = new ButtonBehaviour();
+            buttonBehaviour.OnAdOpening += OnAdOpening;
+        }
+
+        // Creates a rewarded ad.
+        public void CreateRewardedAd(string adUnitId)
+        {
+            Debug.Log("Dummy " + MethodBase.GetCurrentMethod().Name);
+        }
+
+        // Load a rewarded ad.
+        public void LoadAd(AdRequest request)
+        {
+            Debug.Log("Dummy " + MethodBase.GetCurrentMethod().Name);
+            if (Screen.width > Screen.height) //Landscape
+            {
+                if (Screen.width >= 1080)
+                {
+                    LoadAndSetPrefabAd(prefabAds[new AdSize(1024, 768)]);
+                }
+                else
+                {
+                    LoadAndSetPrefabAd(prefabAds[new AdSize(480, 320)]);
+                }
+            }
+            else
+            {
+                if (Screen.height < 1080)
+                {
+                    LoadAndSetPrefabAd(prefabAds[new AdSize(320, 480)]);
+                }
+                else
+                {
+                    LoadAndSetPrefabAd(prefabAds[new AdSize(768, 1024)]);
+                }
+            }
+
+            if (prefabAd != null)
+            {
+                if (OnAdLoaded != null)
+                {
+                    OnAdLoaded.Invoke(this, EventArgs.Empty);
+                }
+            }
+            else
+            {
+                if (OnAdFailedToLoad != null)
+                {
+                    OnAdFailedToLoad.Invoke(this, new AdErrorEventArgs()
+                    {
+                        Message = "Prefab Ad is Null"
+                    });
+                }
+            }
+        }
+
+        // Determines whether the rewarded ad has loaded.
+        public bool IsLoaded()
+        {
+            Debug.Log("Dummy " + MethodBase.GetCurrentMethod().Name);
+            if (prefabAd != null)
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        // Returns the mediation adapter class name.
+        public string MediationAdapterClassName()
+        {
+            return new ResponseInfoDummyClient().GetMediationAdapterClassName();
+        }
+
+        // Returns the reward item for the loaded rewarded ad.
+        public Reward GetRewardItem()
+        {
+            Debug.Log("Dummy " + MethodBase.GetCurrentMethod().Name);
+            return new Reward()
+            {
+                Type = "Dummy Reward",
```

Comment:
I'd like to start moving away from using "Dummy" in these clients, since they are more than just stubs. Let's just call this "Reward" and remove "Dummy"